### PR TITLE
Add `tg4-group` contract

### PR DESCRIPTION
**Pull request by maurolacy**
Based on `cw4-group`. Adapted to `tgrade-bindings` (`TgradeMsg`), and `tgrade-utils` hooks.

~~Publishing as Draft, as tests are still failing.~~ (fixed)

### CODE REVIEW

File: contracts/tg4-group/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+[package]
+name = "tg4-group"
+version = "0.6.0"
+authors = ["Ethan Frey <ethanfrey@users.noreply.github.com>"]
```
Comment (ueco-jb) [inline contracts/tg4-group/Cargo.toml:4]: Don't be so modest!
```suggestion
authors = ["Mauro Lacy <mauro@confio.gmbh>"]
```

File: contracts/tg4-group/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+edition = "2018"
+description = "Simple tg4 implementation of group membership controlled by admin"
+license = "Apache-2.0"
+repository = "https://github.com/CosmWasm/tfi"
```
Comment (ueco-jb) [inline contracts/tg4-group/Cargo.toml:8]:
```suggestion
repository = "https://github.com/confio/poe-contracts"
```

File: contracts/tg4-group/src/contract.rs [hunk 3]
```diff
@@ -0,0 +1,606 @@
+use crate::ContractError::Unauthorized;
```
Comment (ueco-jb) [inline contracts/tg4-group/src/contract.rs:1]: I assume this is mostly copied from cw-plus with `s/weight/points` renamed?

**Review by ueco-jb — APPROVED** [review]
Minor comments.

File: contracts/tg4-group/Cargo.toml (hunk 1 above)
Comment (maurolacy) [reply contracts/tg4-group/Cargo.toml:4]: :-)

File: contracts/tg4-group/src/contract.rs (hunk 3 above)
Comment (maurolacy) [reply contracts/tg4-group/src/contract.rs:1]: That, plus introducing `TgradeMsg` binding. Plus adapting the hooks to tg-utils (based on the tg4-engagement impl)

**Pull request merged**